test/unit/associations/association.test.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+/* jshint -W030 */
+const chai = require('chai');
+const expect = chai.expect;
+const Support = require(__dirname + '/../support');
+const current = Support.sequelize;
+const AssociationError = require(__dirname + '/../../../lib/errors').AssociationError;
+
+describe(Support.getTestDialectTeaser('belongsTo'), function() {
+  it('should throw an AssociationError when two associations have the same alias', function(){
+    const User = current.define('User');
+    const Task = current.define('Task');
+
+    User.belongsTo(Task, { as: 'task' });
+    const errorFunction = User.belongsTo.bind(User, Task, { as: 'task' });
+    const errorMessage = `You have used the alias task in two separate associations. Aliased associations must have unique aliases.`
```

Comment:
Is jshint supposed to catch all of these different types of errors? 
I would think it would at least catch the 2 spaces/space after function parens
Though I'm honestly more familiar with eslint